XmlFormatter/src/Formatter/XmlFormatterProvider.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+using XmlFormatter.src.Enums;
+using XmlFormatter.src.EventMessages;
+using XmlFormatter.src.Interfaces.Formatter;
+
+namespace XmlFormatter.src.Formatter
+{
+    /// <summary>
+    /// A xml formatter class instance
+    /// </summary>
+    class XmlFormatterProvider : IFormatter
+    {
+        /// <inheritdoc/>
+        public event EventHandler<BaseEventArgs> StatusChanged;
+
+        /// <summary>
+        /// The readonly name of this formatter
+        /// </summary>
+        private readonly string name;
+
+        /// <inheritdoc/>
+        public string Name => name;
+
+        /// <summary>
+        /// The extension supported by this formatter
+        /// </summary>
+        private readonly string extension;
+
+        /// <inheritdoc/>
+        public string Extension => extension;
+
+        /// <summary>
+        /// Create a new instance of this class
+        /// </summary>
+        public XmlFormatterProvider()
+        {
+            name = "XmlFormatter";
+            extension = "xml";
+        }
+
+        /// <inheritdoc/>
+        public bool ConvertToFormat(string filePath, string outputName, ModesEnum mode)
+        {
+            switch (mode)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Replace this 'switch' statement with 'if' statements to increase readability.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5053335)